test/trento/application/usecases/clusters_test.exs

```diff
@@ -64,6 +64,7 @@ defmodule Trento.ClustersTest do
       cluster_id = Faker.UUID.v4()
 
       insert(:cluster, id: cluster_id)
+      insert(:cluster, deregistered_at: DateTime.utc_now())
```

Comment:
Personally, i would have added a new test for this. The feature is important enough to have some meaningful test. This looks pretty much hidden here

Reply:
You are right, done